caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -170,48 +174,51 @@ def __del__(self):
     def run(
         self,
         text: str,
-        device: Optional[Union[str, int]] = _DETECT_DEVICE,
-        max_new_tokens=20,
-        min_new_tokens=0,
+        max_new_tokens: Optional[int] = 20,
+        min_new_tokens: Optional[int] = 0,
+        truncate_input_tokens: Optional[int] = 0,
+        decoding_method: Optional[str] = "GREEDY",
+        top_k: Optional[int] = 0,
+        top_p: Optional[float] = 0.0,
```

Comment:
Default `top_p` should also probably be `1.0`, I believe. Otherwise, leaving this unset and setting sampling params will default to greedy decoding

This is also what TGIS does - `1.0` essentially means consider everything: https://github.com/IBM/text-generation-inference/blob/main/router/src/lib.rs#L70C1-L72C8

Reply:
🤔  That make sense. 1.0 and 0.0 are equivalent, so we'll change this

Reply:
It's only equivalent in TGIS! Not in the transformers library

Reply:
That is great to know. 🤔

Reply:
I agree, its probably good to align with defaults in transformers. The good thing is for `top_p` and `typical_p`, the default in transformers is also `1.0`. So the only one that is different is `top_k` which we have as 0 but transformers has as 50